src/test/java/gameframework/game/GameUniverseDefaultTest.java

```diff
@@ -0,0 +1,137 @@
+package gameframework.game;
+
+import gameframework.motion.blocking.MoveBlocker;
+import gameframework.motion.blocking.MoveBlockerChecker;
+import gameframework.motion.blocking.MoveBlockerCheckerDefaultImpl;
+import gameframework.motion.overlapping.OverlapProcessor;
+import gameframework.motion.overlapping.OverlapProcessorDefaultImpl;
+import gameframework.motion.overlapping.Overlappable;
+
+import static org.junit.Assert.assertEquals;
+
+import java.awt.Graphics;
+import java.awt.Point;
+import java.awt.Rectangle;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class GameUniverseDefaultTest {
+
+	protected GameUniverseDefaultImpl game;
+	protected int cptMoveBlockerChecker;
+	protected int cptOverlapProcessor;
+
+	@Before
+	public void init() {
+		this.game = new GameUniverseDefaultImpl();
+		this.game.setGameData(new GameData(new GameConfigurationTest()));
+		this.cptMoveBlockerChecker = 0;
+		this.cptOverlapProcessor = 0;
+	}
+
+	@Test
+	public void testAddAndRemoveGameEntity() {
+		GameEntityMoveBlocker mb = createGameEntityMoveBlocker();
+		GameEntityOverlappable o = createGameEntityOverlappable();
+		this.game.addGameEntity(mb);
+		assertEquals(1, cptMoveBlockerChecker);
+		assertEquals(0, cptOverlapProcessor);
+		this.game.addGameEntity(o);
+		assertEquals(1, cptMoveBlockerChecker);
+		assertEquals(1, cptOverlapProcessor);
+		this.game.removeGameEntity(mb);
+		assertEquals(0, cptMoveBlockerChecker);
+		assertEquals(1, cptOverlapProcessor);
+		this.game.removeGameEntity(o);
+		assertEquals(0, cptMoveBlockerChecker);
+		assertEquals(0, cptOverlapProcessor);
+	}
+	
+	protected GameEntityOverlappable createGameEntityOverlappable() {
+		return new GameEntityOverlappable();
+	}
+	
+	protected GameEntityMoveBlocker createGameEntityMoveBlocker() {
+		return new GameEntityMoveBlocker();
+	}
+	
+	public class GameEntityMoveBlocker implements GameEntity, MoveBlocker{
+		
+		@Override
+		public Rectangle getBoundingBox() {
+			return null;
+		}
+
+		@Override
+		public void draw(Graphics g) {}
+
+		@Override
+		public boolean isMovable() {
+			return false;
+		}
+	}
+	
+	public class GameEntityOverlappable implements GameEntity, Overlappable{
+		
+		@Override
+		public Rectangle getBoundingBox() {
+			return null;
+		}
+
+		@Override
+		public void draw(Graphics g) {}
+
+		@Override
+		public boolean isMovable() {
+			return false;
+		}
+
+		@Override
+		public Point getPosition() {
+			return null;
+		}
+	}
+	
+	// add a count to test
+	public class OverlapProcessorTest extends OverlapProcessorDefaultImpl {
+		@Override
+		public void addOverlappable(Overlappable p) {
+			super.addOverlappable(p);
```

Comment:
do you really need to do this supercall? Is it important that `OverlapProcessorDefaultImpl.addOverlappable` is executed in the context of your unit tests?